src/main/java/com/hackley/cf/demo/Controller/ApplicationUserController.java

```diff
@@ -64,21 +74,33 @@ public String signUpPageSubmit(String username,
     }
 
     @GetMapping("/users/{id}")
-    public String getUsersPage(@PathVariable long id, Model model){
+    public String getUsersPage(@PathVariable long id, Model model, Principal p){
         ApplicationUser user = applicationUserRepository.findById(id).get();
+        List<Post> posts = postRepository.findByCreatorId(user.getId());
         model.addAttribute("user", user);
+        model.addAttribute("posts", posts);
+
+        String princeipal = p == null ? "" : p.getName();
```

Comment:
Having a typo like this will make it harder to remember the variable. Remember that IntelliJ has a refactor that will change a variable's name throughout the entire app